src/azul/service/repository_controller.py

```diff
@@ -35,11 +39,78 @@ def service(self):
     def repository_plugin(cls, catalog: CatalogName) -> RepositoryPlugin:
         return RepositoryPlugin.load(catalog).create(catalog)
 
+    def _parse_range_request_header(self,
+                                    range_specifier: str
+                                    ) -> Sequence[Tuple[Optional[int], Optional[int]]]:
+        """
+        >>> rc = RepositoryController(lambda_context=None, file_url_func=None)
+        >>> rc._parse_range_request_header('bytes=100-200,300-400')
+        [(100, 200), (300, 400)]
+
+        >>> rc._parse_range_request_header('bytes=-100')
+        [(None, 100)]
+
+        >>> rc._parse_range_request_header('bytes=100-')
+        [(100, None)]
+
+        >>> rc._parse_range_request_header('foo=100')
+        []
+
+        >>> rc._parse_range_request_header('')
+        Traceback (most recent call last):
+        ...
+        chalice.app.BadRequestError: BadRequestError: Invalid range specifier ''
+
+        >>> rc._parse_range_request_header('100-200')
+        Traceback (most recent call last):
+        ...
+        chalice.app.BadRequestError: BadRequestError: Invalid range specifier '100-200'
+
+        >>> rc._parse_range_request_header('bytes=')
+        Traceback (most recent call last):
+        ...
+        chalice.app.BadRequestError: BadRequestError: Invalid range specifier 'bytes='
+
+        >>> rc._parse_range_request_header('bytes=100')
+        Traceback (most recent call last):
+        ...
+        chalice.app.BadRequestError: BadRequestError: Invalid range specifier 'bytes=100'
+
+        >>> rc._parse_range_request_header('bytes=-')
+        Traceback (most recent call last):
+        ...
+        chalice.app.BadRequestError: BadRequestError: Invalid range specifier 'bytes=-'
+
+        >>> rc._parse_range_request_header('bytes=--')
+        Traceback (most recent call last):
+        ...
+        chalice.app.BadRequestError: BadRequestError: Invalid range specifier 'bytes=--'
+        """
+
+        def to_int_or_none(value: str) -> Optional[int]:
+            return None if value == '' else int(value)
+
+        parsed_ranges = []
+        try:
+            unit, ranges = range_specifier.split('=')
+            if unit == 'bytes':
+                for range_spec in ranges.split(','):
+                    start, end = range_spec.split('-')
+                    if start == '' and end == '':
+                        raise Exception('Empty range')
+                    parsed_ranges.append((to_int_or_none(start), to_int_or_none(end)))
+            elif unit == '':
+                raise Exception('Empty range unit')
+        except BaseException as e:
```

Comment:
```suggestion
        except Exception as e:
```

Except for ensuring clean-up, we should not catch BaseException.

Reply:
See https://www.python.org/dev/peps/pep-0008/ on BaseException